lang/java/avro/src/main/java/org/apache/avro/io/JsonDecoder.java

```diff
@@ -494,14 +494,23 @@ public Symbol doAction(Symbol input, Symbol top) throws IOException {
         throw error("record-start");
       }
     } else if (top == Symbol.RECORD_END || top == Symbol.UNION_END) {
-      if (in.getCurrentToken() == JsonToken.END_OBJECT) {
+      //AVRO-2034 advance to the end of our object
+      while(in.getCurrentToken() != JsonToken.END_OBJECT){
```

Comment:
what is the fields that are being skipped contain objects ? "field" : { ...} ?

we might need to take in consideration JsonToken.START_OBJECT